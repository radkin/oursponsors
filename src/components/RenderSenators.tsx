import {useEffect} from 'react';
import {getSenators} from '../store/actions/senatorAction';
import {FlatList, StyleSheet, TouchableOpacity, View} from 'react-native';
import RenderRepCard from './RenderRepCard';
import * as React from 'react';
import {responsiveScreenHeight} from 'react-native-responsive-dimensions';
import {useAppDispatch, useAppSelector} from '../hooks';
import { scale } from "react-native-size-matters";

function RenderSenators(props) {
  const dispatch = useAppDispatch();
  const senatorsListData = useAppSelector(state => state.senatorsList);
  const {senators} = senatorsListData;

  useEffect(() => {
    dispatch(getSenators());
  }, [dispatch]);

  const navigation = props.navigation;

  return (
    <FlatList
      data={senators}
      renderItem={({item}) => (
        <TouchableOpacity
          onPress={() =>
            navigation.navigate('Details', {
              value: item,
            })
          }>
          <View style={styles.card}>
            <RenderRepCard value={item} />
          </View>
        </TouchableOpacity>
      )}
      keyExtractor={item => item.id}
      horizontal={false}
    />
  );
}

const styles = StyleSheet.create({
  card: {
    flex: 1,
    borderRadius: 4,
    borderWidth: 2,
    borderColor: '#E8E8E8',
    justifyContent: 'center',
    backgroundColor: 'white',
    height: scale(50),
  },
});
export default RenderSenators;
