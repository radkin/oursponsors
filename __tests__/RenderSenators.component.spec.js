import renderer from 'react-test-renderer';
import RenderSenators from '../src/components/RenderSenators';

describe('RenderSenators', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <RenderSenators/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
