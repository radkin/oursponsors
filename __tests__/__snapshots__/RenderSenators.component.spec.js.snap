// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderSenators renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
